### PR TITLE
refactor(experimental): add codecs-strings package

### PR DESCRIPTION
**Pull request by lorisleiva**
This PR adds string codecs to web3.js. It adds the following functions `getXEncoder`, `getXDecoder` and `getXCodec` for each of the following `X`s:

- `BaseX`
- `BaseXReslice`
- `Base10` (uses `BaseX`)
- `Base16`
- `Base58` (uses `BaseX`)
- `Base64` (uses `atob` and `btoa` on the browser, otherwise uses `BaseXReslice`)
- `Utf8` (uses `TextEncoder` and `TextDecoder`)
- `String` (to support multiple size and encoding strategies)

### CODE REVIEW

**Comment by lorisleiva** [comment]
Current dependencies on/for this PR:
* master
  * **PR #1630** <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/1630" target="_blank">Graphite</a> 
    * **PR #1631** <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/1631" target="_blank">Graphite</a> 
      * **PR #1632** <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/1632" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/1632?utm_source=stack-comment).

File: packages/codecs-strings/src/base64.ts [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+                    // TODO: Coded error.
+                    throw new Error(`Expected a string of base 64, got [${value}].`);
+                }
+            },
+            fixedSize: null,
+            maxSize: null,
+        };
+    }
+
+    return mapEncoder(getBaseXResliceEncoder(alphabet, 6), (value: string): string => value.replace(/=/g, ''));
```
Comment (steveluscher) [inline packages/codecs-strings/src/base64.ts:28]: Following up from the last PR.

* If we're in Node, surely we should use the (presumably) faster `Buffer.from('...', 'base64')` and `buffer.toString('base64')` right?
* You could always test `getBaseXResliceEncoder` separately?

What's the strategy for React Native here?

* Use the `getBaseXResliceEncoder` implementation?
* Get a `if (__REACTNATIVE__)` branch going?
* Do what you did for `text-encoding-impl` and make a `base64-impl` private package.

**Review by steveluscher — APPROVED** [review]

File: packages/codecs-strings/src/base64.ts (hunk 1 above)
Comment (lorisleiva) [reply packages/codecs-strings/src/base64.ts:28]: Sounds good, I think it makes sense to:
- Use `Buffer` for `__NODEJS__`.
- Use `getBaseXResliceEncoder` for `__REACTNATIVE__`.
- Test `getBaseXResliceEncoder` separately.

I'll do that now.

Comment (lorisleiva) [reply packages/codecs-strings/src/base64.ts:41]: Done ✅

**Comment by lorisleiva** [comment]
### Merge Activity

* **Sep 30, 6:16 PM**: @lorisleiva started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/1632).

**Pull request merged**